### PR TITLE
GitHub-Issue#2778: Added README for CloudWatch Logs Sink

### PR DESCRIPTION
**Pull request by MaGonzalMayedo**
### Description
This change introduces the README for the CloudWatch Logs Sink.
 
### Issues Resolved
This PR will work towards resolving CWL-Sink for Data-Prepper [Issue https://github.com/https://github.com/https://github.com/https://github.com/https://github.com/https://github.com/https://github.com/opensearch-project/data-prepper/issues/2778]
Link: https://github.com/opensearch-project/data-prepper/issues/2778#issue-1731228391
 
### Check List
- [x] New functionality includes testing.
- [x] New functionality has been documented.
  - [x] New functionality has javadoc added
- [x] Commits are signed with a real name per the DCO

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/data-prepper/blob/main/CONTRIBUTING.md).

### CODE REVIEW

**Review by dlvenable — DISMISSED** [review]

File: data-prepper-plugins/cloudwatch-logs/README.md [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+```
+pipeline:
+  ...
+  sink:
+    - cloudwatch_logs:
+        aws:
+          region: us-east-1
+          sts_role_arn: arn:aws:iam::123456789012:role/Data-Prepper
+          sts_header_overrides:
```
Comment (markkuhn) [inline data-prepper-plugins/cloudwatch-logs/README.md:19]: all have example data except this? why?

File: data-prepper-plugins/cloudwatch-logs/README.md [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+- `sts_role_arn` (Optional) : The AWS STS role to assume for requests to CloudWatchLogs. which will use the [standard SDK behavior for credentials](https://docs.aws.amazon.com/sdk-for-java/latest/developer-guide/credentials.html). 
+
+- `sts_external_id` (Optional) : The external ID to attach to AssumeRole requests.
+
+## Threshold Configuration
+
+- `batch_size` (Optional) : An integer value that indicates how many events we hold until we make a call to CloudWatch Logs. Defaults to 25.
+
+- `max_event_size` (Optional) : A string representing the max size in bytes of the allowed events. Defaults to "256kb". (Min = "1b", Max = "256 kb")
```
Comment (markkuhn) [inline data-prepper-plugins/cloudwatch-logs/README.md:43]:
```suggestion
- `max_event_size` (Optional) : A string representing the max size in bytes of the allowed events. Defaults to "256kb". (Min = "1b", Max = "256kb")
```

File: data-prepper-plugins/cloudwatch-logs/README.md [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+- `back_off_time` (Optional) : A string representing the amount of time in milliseconds between errored transmission re-attempts. Defaults to "500ms". (Min = "500ms", Max = "1000ms")
+
+## Buffer Type Configuration
+
+- `buffer_type` (Optional) : A string representing the type of buffer to use to hold onto events. Currently only supports `in_memory`.
+
+## Plugin Functionality
+The cloudwatch_logs sink plugin uses credentials to establish a client to CloudWatch Logs. It currently uses the system
+default timestamp for publishing (ex: System.currentTimeMillis()) and also implements an exponential back off strategy
+for retransmission.
```
Comment (markkuhn) [inline data-prepper-plugins/cloudwatch-logs/README.md:60]:
```suggestion
The cloudwatch_logs sink plugin uses credentials to establish a client to CloudWatch Logs. It currently uses the current system timestamp for publishing and implements an exponential back off strategy
for retransmission. 
```

File: data-prepper-plugins/cloudwatch-logs/README.md (hunk 2 above)
Comment (MaGonzalMayedo) [reply data-prepper-plugins/cloudwatch-logs/README.md:43]: thank you, I missed this one.

File: data-prepper-plugins/cloudwatch-logs/README.md (hunk 1 above)
Comment (MaGonzalMayedo) [reply data-prepper-plugins/cloudwatch-logs/README.md:19]: Sorry about that. I missed those values in the original draft. I have added in example values now and another parameter that was missing.

**Review by markkuhn — APPROVED** [review]

File: data-prepper-plugins/cloudwatch-logs/README.md (hunk 1 above)
Comment (dlvenable) [reply data-prepper-plugins/cloudwatch-logs/README.md:19]: We probably don't even need to show this in the example. It is not commonly used.

**Review by dlvenable — APPROVED** [review]

**Review by markkuhn — APPROVED** [review]

**Pull request merged**